q2_moshpit/kraken2/classification.py

```diff
@@ -75,14 +76,22 @@ def get_paths_for_contigs(contig_id, fp):
         return contig_id.rstrip('_contigs'), [fp]
 
     try:
-        if manifest is not None:  # we got reads - use the manifest
+        if manifest is not None:
+            # we got reads - use the manifest
             iterate_over = manifest.iterrows()
             path_function = get_paths_for_reads
         else:
-            iterate_over = (
-                (os.path.basename(fp).split(".")[0], fp)
-                for fp in sorted(glob.glob(os.path.join(seqs.path, "*.fasta")))
+            # we got contigs or MAGs
+            def view_to_paths(view):
+                path = view[0]
+                stem = path.stem
+                absolute = str(pathlib.Path(seqs.path) / path)
+                return stem, absolute
```

Comment:
I don't think the variables are adding a whole lot (maybe `absolute` is, but I dunno if it's worth the line):
```suggestion
                return relpath.stem, str(seqs.path / relpath)
```